From 30989bd0c2aa3f9f4b6f5e393397b39678717f45 Mon Sep 17 00:00:00 2001
From: Charles Keepax <ckeepax@opensource.cirrus.com>
Date: Wed, 20 Nov 2024 10:06:06 +0000
Subject: [PATCH 153/237] sof-soundwire: cs42l43: Correct CapturePCM and
 routing

For headset microphone capture the correct PCM device to use is
"hw:${CardId},1", "hw:${CardId},4" is for built in microphone capture.

Adjust the routing as well since :0,1 is connected to data port 2 on the
codec.

Closes: https://github.com/alsa-project/alsa-ucm-conf/pull/466
Signed-off-by: Charles Keepax <ckeepax@opensource.cirrus.com>
Signed-off-by: Jaroslav Kysela <perex@perex.cz>
Origin: upstream, https://github.com/alsa-project/alsa-ucm-conf/commit/30989bd0c2aa3f9f4b6f5e393397b39678717f45
Bug-Ubuntu: https://bugs.launchpad.net/bugs/2106394
Signed-off-by: Atlas Yu <atlas.yu@canonical.com>
---
 ucm2/sof-soundwire/cs42l43.conf | 10 +++++-----
 1 file changed, 5 insertions(+), 5 deletions(-)

diff --git a/ucm2/sof-soundwire/cs42l43.conf b/ucm2/sof-soundwire/cs42l43.conf
index dc11840..4b4e118 100644
--- a/ucm2/sof-soundwire/cs42l43.conf
+++ b/ucm2/sof-soundwire/cs42l43.conf
@@ -28,19 +28,19 @@ SectionDevice."Headset" {
 		cset "name='cs42l43 ADC1 Input' 'IN1'"
 		cset "name='cs42l43 Decimator 1 Mode' 'ADC'"
 
-		cset "name='cs42l43 DP1TX1 Input' 'Decimator 1'"
-		cset "name='cs42l43 DP1TX2 Input' 'Decimator 1'"
+		cset "name='cs42l43 DP2TX1 Input' 'Decimator 1'"
+		cset "name='cs42l43 DP2TX2 Input' 'Decimator 1'"
 	]
 
 	DisableSequence [
 		cset "name='cs42l43 Decimator 1 Switch' 0"
-		cset "name='cs42l43 DP1TX1 Input' 'None'"
-		cset "name='cs42l43 DP1TX2 Input' 'None'"
+		cset "name='cs42l43 DP2TX1 Input' 'None'"
+		cset "name='cs42l43 DP2TX2 Input' 'None'"
 	]
 
 	Value {
 		CapturePriority 200
-		CapturePCM "hw:${CardId},4"
+		CapturePCM "hw:${CardId},1"
 		CaptureMixer "default:${CardId}"
 		CaptureMixerElem "cs42l43 Headset Microphone"
 		JackControl "Headset Mic Jack"
-- 
2.47.2

